tools/c7n_azure/tests/test_dependency_manager.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2015-2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import filecmp
+import os
+import re
+import shutil
+import sys
+import tempfile
+import unittest
+
+from azure_common import BaseTest
+from c7n_azure.dependency_manager import DependencyManager
+
+
+@unittest.skipIf(sys.version_info < (3, 6), "Functions is not supported in this version")

```

Comment:
what's python 3 specific about the dep manager code and tests?

Reply:
Nothing that cannot be fixed: 
 - `SRE_Match` (`regex match`) object in python 2.7 don't implement `__getitem__()`
 - `json.dumps` in Python 2.7 dumps json with a different fields order, so you can't just compare two files.

`dependency_manager` is a part of Azure Functions mode which is not supported with Python 2.7, so I don't think there is a real need to test it.

Reply:
1. SRE -> `.items()[0]`
2. loading and deserializing the file contents would suffice. alternatively dumping using an ordered dict.

afaicr part of the intent with dependency manager is that a 2.7 install would be able to deploy an azure function, its not a question of the azure function runtime version, but the cli execution/provision compatibility which is what this is testing.

Reply:
You're right about the intent, however we saw some weird issues trying to make all dependencies with Python 2.7 to work. I honestly don't remember what it was, I think it was some weirdness with some packages without wheels.

So, we decided not to invest in 2.7 support for functions mode until we hear real ask for it because it is not something that needs to be supported going forward.

As I said, nothing that cannot be fixed (at least in this PR) :)